hs/src/Reach/Report.hs

```diff
@@ -1,22 +1,44 @@
-module Reach.Report (Report, startReport) where
+module Reach.Report (Report, startReport, CompileErrorException(..)) where
 
 import Control.Concurrent
 import Control.Concurrent.Async
 import Control.Exception
 import Control.Monad.Reader
 import Data.Aeson
 import Data.Time
+import Data.Typeable (cast)
 import Network.HTTP.Client.Conduit (httpNoBody)
 import Network.HTTP.Client.TLS
 import Network.HTTP.Conduit
-import Network.HTTP.Simple (setRequestBodyJSON, setRequestMethod)
+import Network.HTTP.Simple ({-setRequestBodyJSON, Remove comment braces when done -} setRequestMethod)
 import Reach.CommandLine
 import Reach.Version
 import System.Environment
 
+-- Remove when finished
+import GHC.Exts
+import Data.List (intercalate)
+
 --- TODO maybe have each part collect some information and report it back through a (Map String String)
 type Report = Either SomeException ()
 
+data CompileErrorException = CompileErrorException
+  { cee_json :: String
+  , cee_pretty :: String
+  }
+
+instance Show CompileErrorException where
+  show = cee_pretty
```

Comment:
This is actually wrong and we do need that unsafe thing, because it exists because the IDE wants to see errors as JSON.